kotlintest-extensions/kotlintest-extensions-spring/src/test/kotlin/io/kotlintest/spring/SpringTestExecutionListenerTest.kt

```diff
@@ -0,0 +1,69 @@
+package io.kotlintest.spring
+
+import io.kotlintest.extensions.TestListener
+import io.kotlintest.shouldBe
+import io.kotlintest.specs.FunSpec
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.test.context.TestContext
+import org.springframework.test.context.TestExecutionListeners
+import org.springframework.test.context.TestExecutionListener as SpringTestExecutionListener
+
+@TestExecutionListeners(DummyTestExecutionListener::class, mergeMode = TestExecutionListeners.MergeMode.MERGE_WITH_DEFAULTS)
+@SpringBootTest(classes = [Components::class])
+class SpringTestExecutionListenerTest : FunSpec() {
+
+  @Autowired
+  lateinit var userService: UserService
+
+  override fun listeners(): List<TestListener> {
+    return listOf(SpringListener)
+  }
+
+  init {
+    test("Should autowire with spring listeners") {
+      userService.repository.findUser()
+    }
+
+    test("Should execute Spring Listeners") {
+      // 2 before/after spec, 4 before/after test, 2 test execution
+      DummyTestExecutionListener.count shouldBe 8
+    }
+  }
+
+}
+
+class DummyTestExecutionListener : SpringTestExecutionListener {
+
+  override fun beforeTestExecution(testContext: TestContext) {
+    count++
+  }
+
+  override fun afterTestClass(testContext: TestContext) {
+    count++
+  }
```

Comment:
Each of these counts should be separate otherwise we don't know if the correct ones were invoked just that the correct quantity was.

Reply:
Good one. I made the change and found a bug because of it.